wandb/apis/public/api.py

```diff
@@ -1048,3 +1048,30 @@ def list_jobs(self, entity, project):
             return [x["node"]["artifacts"] for x in artifacts]
         except requests.exceptions.HTTPError:
             return False
+
+    @normalize_exceptions
+    def exists(self, name, type=None):
```

Comment:
Also, @moredatarequired --- isn't this name too general? AFAIK, `Api` isn't an artifacts-specific class. Maybe this should be named `artifact_exists()` instead?

Reply:
The intention was that it would be expanded to check if runs exist, etc. That's why it currently raises a `NotImplementedError` rather than a `ValueError` if the input string can't be parsed as an artifact.

Reply:
Do we use separate namespaces for these objects? Like could an artifact be named the same as a run? Or could a new object be introduced in the future without a guarantee of having disjoint namespaces?

If it's guaranteed that the kind of object can be inferred from the name, now and in the future, then this sounds good to me!

Reply:
I guess another question is whether the `type` parameter would really apply to these other objects. Maybe it should be named `artifact_type` for clarity?

Though that takes me back to maybe this should be `artifact_exists(name, type)` instead of `exists(name, artifact_type)`. I assume you've thought it all through; I'm just thinking as a user I've found these super-methods with parameters that only apply sometimes hard to reason about before (e.g. consider the learning curve for Pandas).

Reply:
also looking at the name of the PR it is explicitly says check if artifact exists, so to me it sounds it either should be on an artifact or name-spaced with artifact. 
totally agree with Tima!

Reply:
and like Tima said the api doesn't make sense for general object check

Reply:
changed this to `artifact_exists` and `artifact_collection_exists`!